test/util/opal_json.c

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright (c) 2024      Amazon.com, Inc. or its affiliates.
+ *                         All Rights reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+
+#include "opal_config.h"
+#include "opal/util/json/opal_json.h"
+#include "opal/runtime/opal.h"
+#include "support.h"
+#include <stdio.h>
+#include <string.h>
+
+enum INPUT_TYPE {
+    JSON_STRING,
+    JSON_FILE,
+    INPUT_TYPE_COUNT,
+};
+
+#define FREE_JSON(obj)          \
+    do {                        \
+        opal_json_free(&(obj)); \
+        obj = NULL;             \
+    } while (0)
+
+static int load_json(const char *string, enum INPUT_TYPE input_type, const opal_json_t **json)
+{
+    int ret = 0;
+    char *filename = "tmp.json";
```

Comment:
I think using "tmp.json" is dangerous.  I am sure I have several tmp.json files sitting around, and I would be surprised if suddenly one of them changed or disappeared.

Suggest:
```
char filename[32];
strdup(filename, "tmpfile.XXXXXX");
mktemp(filename);
....
```

Reply:
Addressed in update thanks!